server/galaxyls/tests/files/simple_params_01.xml

```diff
@@ -1,15 +1,15 @@
-<tool id="test01" name="Sample Test Generation">
+<tool id="simple_params_01" name="Simple param inputs">
     <inputs>
+        <param argument="--arg-01" type="integer"/>
+        <param argument="-arg-02" type="float"/>
         <param name="param-01" type="text"/>
```

Comment:
So a `name` can not contain dashes, as in cheetah (the command section) it would become `$param-01` and this is not allowed in Python. No dashes in variable names. So all `names` need to use `_` for example.

However, on a UNIX CLI you can have something like `--param-01` ... so an `argument="--param-01"` is valid. Galaxy will convert this internally to (the `name`) `param_01` so it will become `$param_01`.

Does that make sense?

Reply:
Absolutely :)
I wonder if then, the [type of the `name` attribute in the XSD](https://github.com/galaxyproject/galaxy/blob/52ef6c812f034cf9468c6bc8896409e5e783ece0/lib/galaxy/tool_util/xsd/galaxy.xsd#L2672) should be a `simpleType` with a `pattern` restriction to allow only valid names instead of `xs:string`. That way it will be marked as an error when validating the file. Or we can just add an additional validation for names directly in the `language-server`.

Reply:
XSD is the correct way here I guess! That's an awesome idea. We need to make the XSD stricter.